tofhir-engine/src/main/scala/io/tofhir/engine/data/write/SinkHandler.scala

```diff
@@ -40,44 +42,29 @@ object SinkHandler {
           // handle the exception caused by invalid mapping results
           t.getCause match {
             case e:FhirMappingInvalidResourceException =>
-              // log the mapping job results including the number of invalid resources provided by FhirMappingInvalidResourceException
-              val jobResult = FhirMappingJobResult(mappingJobExecution, mappingUrl, numOfInvalids, numOfNotMapped, numOfFhirResources, e.getNumOfInvalidResources)
-              logger.info(jobResult.toLogstashMarker, jobResult.toString)
+              logMappingJobResult(mappingJobExecution,mappingUrl,numOfFhirResources,e.getProblems,mappingErrors,invalidInputs)
           }
           throw t
         }
       }
-      //Get the not written resources
-      val notWrittenResources = fhirWriteProblemsAccum.value
-      val numOfNotWritten = notWrittenResources.size()
-
-      //Log the job result
-      val jobResult = FhirMappingJobResult(mappingJobExecution, mappingUrl, numOfInvalids, numOfNotMapped, numOfFhirResources, numOfNotWritten)
-      logger.info(jobResult.toLogstashMarker, jobResult.toString)
-
-      //Log the mapping errors
-      if (numOfNotMapped > 0 || numOfInvalids > 0) {
-        mappingErrors.union(invalidInputs).foreach(r =>
-          logger.warn(r.toLogstashMarker, r.toString)
-        )
-      }
-      if (numOfNotWritten > 0)
-        notWrittenResources.forEach(r =>
-          logger.warn(r.toLogstashMarker, r.toString)
-        )
+      logMappingJobResult(mappingJobExecution,mappingUrl,numOfFhirResources,fhirWriteProblemsAccum.value,mappingErrors,invalidInputs)
       //Unpersist the data frame
       df.unpersist()
     } catch {
       case t:Throwable => {
         t.getCause match {
           // FhirMappingInvalidResourceException is already handled above
           case e:FhirMappingInvalidResourceException => None
+          // FhirMappingException is already handled and logged by Spark while running the mapping
```

Comment:
What do you mean by "already handled and logged by Spark"?

Reply:
When FhirMappingException is thrown, Spark will encapsulate it with SparkException and prints its log. Therefore, we do not need to handle this exception here, otherwise we end up with duplicate logs. Therefore, I simply skipped these exceptions which are already logged by Spark.